src/main/java/Parser.java

```diff
@@ -0,0 +1,159 @@
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Duke's Parser. Deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Takes in the user's command and decides the appropriate action to take, then returns the output string for
+     * the UI object to display. Uses references to taskList, storage, scanner and ui in order to edit the task list,
+     * store the task list when done, scan for arguments to commands and end the ui when needed.
+     * @param nextCommand The next command for the parser to parse.
+     * @param taskList The TaskList to be edited.
+     * @param storage The Storage object that deals with the data file.
+     * @param sc The provided Scanner to read command arguments.
+     * @param ui The ui that called this parser. Referenced to end ui when needed.
+     * @return Output String for the associated UI object to display.
+     */
+    public String parse(String nextCommand, TaskList taskList, Storage storage, Scanner sc, Ui ui) {
+        StringBuilder outputString = new StringBuilder();
+
+        switch (nextCommand) {
+        case "list":
+            outputString.append("    ____________________________________________________________\n");
```

Comment:
might give StringBuilder a function to handle adding line?

Reply:
StringBuilder is a java class. I'm not sure I can add to that?